cr.c

```diff
@@ -84,9 +84,11 @@ static void dill_ctx_timer_handler(int sig, siginfo_t *si, void *uc) {
 #ifndef __APPLE__
     struct dill_ctx_cr *ctx = si->si_value.sival_ptr;
     int or = timer_getoverrun(ctx->timer);
-    if(or) ctx->do_poll = 1;
+    if(or) {
+        __sync_lock_test_and_set(&ctx->do_poll, 1);
+    }
 #else
-    dill_poll_count++;
+    __sync_fetch_and_add(&dill_poll_count, 1);
```

Comment:
Not synchronized TSC doesn't matter: If thread is migrated from one core the TSC can:

1. Jump forward -- the worse thing that can happen is that clock_gettime()  will be called immediately rather than later on.
2. Jump backward to before the last recorded value: We can simply do clock_gettime() in this case.
3. Jump backwards a bit, so that the new value is still greater than last recorded value. In this case the time may stand still for a while, but only up to 1ms.